internal/app/networkd/pkg/reg/reg.go

```diff
@@ -155,16 +156,54 @@ func toCIDR(family uint8, prefix net.IP, prefixLen int) string {
 }
 
 // Check implements the Health api and provides visibilty into the state of networkd.
-// Ready signifies the daemon (api) is healthy and ready to serve requests.
+// We determine networkd health based on neighbor (arp table) entries.
+// Under normal operating circumstances, there should be a steady state of reachable neighbor entries.
+// If we get in a situation where all of our neighbor entries are stale, then we're in trouble.
 func (r *Registrator) Check(ctx context.Context, in *empty.Empty) (reply *healthapi.HealthCheckResponse, err error) {
+	// Set initial state to unknown
 	reply = &healthapi.HealthCheckResponse{
 		Messages: []*healthapi.HealthCheck{
 			{
-				Status: healthapi.HealthCheck_SERVING,
+				Status: healthapi.HealthCheck_UNKNOWN,
 			},
 		},
 	}
 
+	var neighbors []rtnetlink.NeighMessage
+
+	neighbors, err = r.Conn.Neigh.List()
```

Comment:
never checked that, but if this socket-like thing, would it be bad if we do concurrent requests to it (from API and networkd doing its job)

Reply:
https://github.com/mdlayher/netlink/blob/master/conn.go#L18-L20
https://github.com/mdlayher/netlink/blob/master/conn.go#L116-L118

(rtnetlink uses netlink under the covers )
From what I gather each netlink request acquires a lock in the kernel, so I don't believe this is a concern. 
We _may_ want to do some sort of pooling for these checks, but I'm not sure we're at a high enough threshold per connection. I'm not sure what the value of the threshold would be ( multiple a second?), but we're currently only at 1 per healthcheck interval + whatever an end user would add on top of that.

Reply:
actually `Execute()` (method behind all the higher-level functions) does locking: https://github.com/mdlayher/netlink/blob/master/conn.go#L122, so we should be good